tests/test_reduction_ops.py

```diff
@@ -233,6 +233,32 @@ def test_accuracy_cummin(shape, dtype):
 
 NONZERO_SHAPES = [(2, 32)] if QUICK_MODE else REDUCTION_SHAPES + [(2637,)]
 
+# TODO: failed at torch.bfloat16
+CUMPROD_DTYPES = [torch.float32, torch.float64]
+CUMPROD_SHAPES = (
+    [(1, 32)] if QUICK_MODE else REDUCTION_SHAPES + [(2637,), (16, 1025, 255)]
+)
+
+
+@pytest.mark.cumprod
+@pytest.mark.parametrize("shape", CUMPROD_SHAPES)
+@pytest.mark.parametrize("dtype", CUMPROD_DTYPES)
+def test_accuracy_cumprod(shape, dtype):
+    dim = 1 if shape == REDUCTION_SHAPES[-1] else -1
+
+    if dtype in INT_DTYPES:
+        inp = torch.randint(-3, 3, shape, device=flag_gems.device).to(dtype)
+    else:
+        inp = torch.randn(shape, dtype=dtype, device=flag_gems.device)
+
+    ref_inp = to_reference(inp, True)
+    ref_out = torch.cumprod(ref_inp, dim=dim)
+
+    with flag_gems.use_gems():
+        res_out = torch.cumprod(inp, dim=dim)
```

Comment:
If not registered in aten library, your implementation of cumprod won't be called by torch dispatch.

Reply:
You could refer to this file https://github.com/FlagOpen/FlagGems/blob/master/src/flag_gems/__init__.py , and I think your operator cov faces the same problem.